### PR TITLE
upgrade clarity components to 1.0

### PR DESCRIPTION
**Pull request by jwangyangls**
Signed-off-by: Yogi_Wang <Yogiwang@Yogi-WangdeMacBook-Pro.local>

### CODE REVIEW

File: src/portal/lib/src/config/system/system-settings.component.ts [hunk 1]
```diff
@@ -84,6 +84,7 @@ export class SystemSettingsComponent implements OnChanges {
     }
 
     setRepoReadOnlyValue($event: any) {
+        console.log(1111)
```
Comment (jwangyangls) [inline src/portal/lib/src/config/system/system-settings.component.ts]: remove console log

File: src/portal/lib/src/log/recent-log.component.spec.ts [hunk 2]
```diff
@@ -143,13 +149,17 @@ describe('RecentLogComponent (inline template)', () => {
...
+        // expect(component.recentLogs).toBeTruthy();
+        // expect(component.logsCache).toBeTruthy();
+        // expect(component.recentLogs.length).toEqual(3);
+        
       });
     });
   }));
 
-  it('should support filtering list by keywords', async(() => {
+  fit('should support filtering list by keywords', async(() => {
```
Comment (jwangyangls) [inline src/portal/lib/src/log/recent-log.component.spec.ts]: remove f

File: src/portal/lib/src/repository-gridview/repository-gridview.component.spec.ts [hunk 3]
```diff
@@ -155,7 +155,7 @@ describe('RepositoryComponentGridview (inline template)', () => {
     });
   }));
 
-  // Will fail after upgrade to angular 6. todo: need to fix it.
+  // Will fail after upgrade to angular 6. todo: need to fix it. //xit
```
Comment (jwangyangls) [inline src/portal/lib/src/repository-gridview/repository-gridview.component.spec.ts]: remove //xit

File: src/portal/src/app/config/auth/config-auth.component.html [hunk 4]
```diff
@@ -30,8 +30,10 @@
                 [disabled]="!currentConfig.uaa_verify_cert.editable">
         </div>
         <div class="form-group">
-            <label for="uaa-cert" class="required">{{'CONFIG.UAA.VERIFY_CERT' | translate}}</label>
-            <clr-checkbox id="uaa-cert" name="uaa-cert" [(clrChecked)]="currentConfig.uaa_verify_cert.value" [clrDisabled]="!currentConfig.uaa_verify_cert.editable"></clr-checkbox>
+            <clr-checkbox-wrapper>
+                    <input type="checkbox" clrCheckbox name="uaa-cert" id="uaa-cert" required [(ngModel)]="currentConfig.uaa_verify_cert.value" [disabled]="!currentConfig.uaa_verify_cert.editable" />
+                    <label for="uaa-cert" class="required">{{'CONFIG.UAA.VERIFY_CERT' | translate}}</label>
+                  </clr-checkbox-wrapper>
```
Comment (jwangyangls) [inline src/portal/src/app/config/auth/config-auth.component.html]: format error

File: src/portal/src/app/config/auth/config-auth.component.html [hunk 5]
```diff
@@ -191,27 +193,33 @@
...
+                    <input type="checkbox" clrCheckbox name="selfReg" id="selfReg" [(ngModel)]="currentConfig.self_registration.value" [disabled]="disabled(currentConfig.self_registration)" />
+                    <label>
+                            <a href="javascript:void(0)" role="tooltip" aria-haspopup="true" class="tooltip tooltip-top-right top-7">
+                                    <clr-icon shape="info-circle" class="info-tips-icon" size="24"></clr-icon>
+                                    <span *ngIf="checkable; else elseBlock" class="tooltip-content">{{'CONFIG.TOOLTIP.SELF_REGISTRATION_ENABLE' | translate}}</span>
+                                    <ng-template #elseBlock>
+                                        <span class="tooltip-content">{{'CONFIG.TOOLTIP.SELF_REGISTRATION_DISABLE' | translate}}</span>
+                                    </ng-template>
+                                </a>
+                    </label>
```
Comment (jwangyangls) [inline src/portal/src/app/config/auth/config-auth.component.html]: format

File: src/portal/src/app/config/auth/config-auth.component.html [hunk 6]
```diff
@@ -191,27 +193,33 @@
...
+            <clr-checkbox-wrapper>
+                    <input type="checkbox" clrCheckbox name="ldapVerifyCert" id="ldapVerifyCert" [ngModel]="currentConfig.ldap_verify_cert.value" [disabled]="disabled(currentConfig.ldap_scope)"
+                    (ngModelChange)="setVerifyCertValue($event)" />
+                    <label>
+                            <a href="javascript:void(0)" role="tooltip" aria-haspopup="true" class="tooltip tooltip-top-right top-7">
+                                    <clr-icon shape="info-circle" class="info-tips-icon" size="24"></clr-icon>
+                                    <span class="tooltip-content">{{'CONFIG.TOOLTIP.VERIFY_CERT' | translate}}</span>
+                            </a>
+                    </label>
+                </clr-checkbox-wrapper>
```
Comment (jwangyangls) [inline src/portal/src/app/config/auth/config-auth.component.html]: format

File: src/portal/src/app/dev-center/dev-center.component.scss [hunk 7]
```diff
@@ -5,4 +5,5 @@
     bottom: 0;
     left: 0;
     right: 0;
+    overflow:scroll
```
Comment (jwangyangls) [inline src/portal/src/app/dev-center/dev-center.component.scss]: missing ;

File: src/portal/src/app/project/project.component.html [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <div class="row">
-    <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12">
+    <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12 positionR">
```
Comment (jwangyangls) [inline src/portal/src/app/project/project.component.html]: css style use -

File: src/portal/src/app/user/user.component.scss [hunk 9]
```diff
@@ -37,5 +37,7 @@
     visibility: hidden !important;
 }
 .rightPos {
-    position: absolute; right: 20px; margin-top: -7px; height:32px; z-index: 100;
+    position: absolute; right: 20px; 
+    // margin-top: 0px; 
+    height:32px; z-index: 100;
```
Comment (jwangyangls) [inline src/portal/src/app/user/user.component.scss]: use a new line

File: tests/resources/Harbor-Pages/HomePage_Elements.robot [hunk 10]
```diff
@@ -24,5 +24,6 @@ ${realname_xpath}  //*[@id="realname"]
 ${newPassword_xpath}  //*[@id="newPassword"]
 ${confirmPassword_xpath}  //*[@id="confirmPassword"]
 ${comment_xpath}  //*[@id="comment"]
-${signup_xpath}  /html/body/harbor-app/harbor-shell/clr-main-container/div/div/sign-in/sign-up/clr-modal/div/div[1]/div/div[1]/div/div[3]/button[2]
+${signup_xpath}  //clr-modal/div/div[1]/div/div/div[3]/button[2]
+# ${signup_xpath}  /html/body/harbor-app/harbor-shell/clr-main-container/div/div/sign-in/sign-up/clr-modal/div/div[1]/div/div[1]/div/div[3]/button[2]
```
Comment (jwangyangls) [inline tests/resources/Harbor-Pages/HomePage_Elements.robot]: remove the comment

File: tests/resources/Harbor-Pages/Vulnerability.robot [hunk 11]
```diff
@@ -24,8 +24,13 @@ Set Vulnerabilty Serverity
     [Arguments]  ${level}
     Goto Project Config
     #enable first
-    Click Element  //project-detail//clr-checkbox[@name="prevent-vulenrability-image"]//label
-    Checkbox Should Be Selected  //project-detail//clr-checkbox//input[@name="prevent-vulenrability-image"]
+<<<<<<< HEAD
```
Comment (jwangyangls) [inline tests/resources/Harbor-Pages/Vulnerability.robot]: resolve the conflict

File: tests/resources/Harbor-Pages/Project-Helmcharts_Elements.robot [hunk 12]
```diff
@@ -29,8 +29,8 @@ ${detail_value}  //*[@id="value-link"]
 ${value_content}  //*[@id="value-content"]/hbr-chart-detail-value
 
 ${version_bread_crumbs}  //project-chart-detail//a[contains(.,"Versions")]
-${version_checkbox}  //list-chart-version//clr-dg-cell[1]/clr-checkbox
+${version_checkbox}//clr-dg-row//clr-checkbox-wrapper/input
```
Comment (zhoumeina) [inline tests/resources/Harbor-Pages/Project-Helmcharts_Elements.robot]: add the whitespace back

File: tests/resources/Harbor-Pages/Project-Helmcharts_Elements.robot [hunk 13]
```diff
@@ -29,8 +29,8 @@ ${detail_value}  //*[@id="value-link"]
...
 ${version_delete}  //clr-dg-action-bar/button[contains(.,"DELETE")]
-${version_confirm_delete}  //clr-modal//button[contains(.,"DELETE")]
+${version_confirm_delete}  //clr-modal//button[contains(.,"çDELETE")]
```
Comment (zhoumeina) [inline tests/resources/Harbor-Pages/Project-Helmcharts_Elements.robot]: çDELETE  is a typo?

File: src/portal/lib/src/config/system/system-settings.component.html [hunk 14]
```diff
@@ -38,18 +40,25 @@
...
-                    <span class="tooltip-content">{{'CONFIG.TOOLTIP.REPO_TOOLTIP' | translate}}</span>
-                </a>
-            </clr-checkbox>
+                <label for="repoReadOnly">{{'CONFIG.REPO_READ_ONLY' | translate}}</label>
+                <clr-checkbox-wrapper>
+                    <input type="checkbox" clrCheckbox name="repoReadOnly" id="repoReadOnly"  [ngModel]="systemSettings.read_only.value"
+                    (ngModelChange)="setRepoReadOnlyValue($event)" />
+                    <label>
+                        <a href="javascript:void(0)" role="tooltip" aria-haspopup="true" class="tooltip tooltip-top-right read-tooltip"
+                        >
```
Comment (zhoumeina) [inline src/portal/lib/src/config/system/system-settings.component.html]: don't format in this way , keep in one line.

**Pull request merged**